### ISSUE TITLE
From Cloudformation documentation, tags should be expressed as arrays of Key/Value

### ISSUE THREAD

**Issue by gautaz** [aws-cloudformation/cfn-lint#3719]
### CloudFormation Lint Version

cfn-lint 1.10.0

### What operating system are you using?

Linux hepao 6.6.49-NixOS SMP PREEMPT_DYNAMIC Wed Sep  4 11:28:31 UTC 2024 x86_64 GNU/Linux

### Describe the bug

Looking at [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html) regarding tags on, for instance, Batch job definitions, a template for this type of resource would be (from my understanding):

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  TesJobDefinition:
    Properties:
      Tags:
        - Key: key
          Value: value
      Type: Container
    Type: AWS::Batch::JobDefinition
```

But `cfn-lint` validation fails:

```console
❯ cfn-lint test.yaml
E3012 [{'Key': 'key', 'Value': 'value'}] is not of type 'object', 'string'
/workspace/test.yaml:5:7
```

### Expected behavior

The expectation would be that the example template is valid.

Nevertheless, I must say that I find the Cloudformation documentation on this point particularly awkward as I would have expected an object to express a list of tags.
As a matter of fact, this is what `cfn-lint` currently considers as valid:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  TesJobDefinition:
    Properties:
      Tags:
        key: value
      Type: Container
    Type: AWS::Batch::JobDefinition
```

But as the current CloudFormation documentation goes, other tools like [cloudtools/troposphere](https://github.com/cloudtools/troposphere/tree/main) seem to follow it and generate tags as arrays of Key/Value.

Looking at PRs related to this, I have stumbled upon #3437 which seems to say that the Cloudformation specification says "object" and the documentation says "array".
As troposphere also uses the Cloudformation specification to generate most of its code, I am rather puzzled...

So, which one is right?
And where is the source of truth if not the Cloudformation documentation?

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  TesJobDefinition:
    Properties:
      Tags:
        - Key: key
          Value: value
      Type: Container
    Type: AWS::Batch::JobDefinition
```

**Comment by kddejong** [comment]
If you use your template against the service early validation will fail with:
```
Property validation failure: [Value of property {/Tags} does not match type {Map}]
```

From this I think you need to follow the Tag structure of mapping like `Key: Value`

**Comment by gautaz** [comment]
Hello @kddejong, thanks for your input.

I am not sure about what you call the "service early validation", I suppose this is related to the CLI command [`aws cloudformation validate-template`](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/validate-template.html).

So I have tested it with the following template:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  TesJobDefinition:
    Properties:
      Tags:
        - Key: key
          Value: value
      Type: Container
    Type: AWS::Batch::JobDefinition
```

And the result seems to be a success (0 status code):

```console
❯ aws cloudformation validate-template --template-body file://test.yaml
{
    "Parameters": []
}
```

Can you give me a bit more details so that I am sure that we are talking about the same validation process?

Also the comments I have highlighted in the issue description make the distinction between what is said in the Clouformation documentation and the Cloudformation specification. Where is this specification available if this is not the documentation?

**Comment by kddejong** [comment]
Early validation is where the property values are validated against the [provider schema](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html)

https://aws.amazon.com/about-aws/whats-new/2024/03/aws-cloudformation-new-validation-checks-stack-operations/

To use this feature you have to create a stack with that template.

**Comment by gautaz** [comment]
OK, so looking at the provider schema (eu-west-3 in my case) for `AWS::Batch::JobDefinition`, the property `Tags` is described this way:

```json
    "Tags" : {
      "type" : "object"
    }
```

Now I understand your point, so the issue is troposphere generating an array of Key/Value instead of a dictionary.
Thanks @kddejong for the clarification, I can close the issue.
